src/bin/set-version/main.rs

```diff
@@ -75,11 +79,12 @@ fn process(args: Args) -> Result<()> {
     if all {
         deprecated_message("The flag `--all` has been deprecated in favor of `--workspace`")?;
     }
-    let all = workspace || all;
+    let all = workspace || all || LocalManifest::find(&None)?.is_virtual();
+
     let manifests = if all {
         Manifests::get_all(&manifest_path)
-    } else if let Some(ref pkgid) = pkgid {
-        Manifests::get_pkgid(manifest_path.as_deref(), pkgid)
+    } else if let Some(id) = pkgid.get(0) {
+        Manifests::get_pkgid(manifest_path.as_deref(), id)
     } else {
         Manifests::get_local_one(&manifest_path)
     }?;
```

Comment:
The intention of using `clap-cargo` is we defer to it for interpreting the combination of flags.

See https://docs.rs/clap-cargo/0.6.1/clap_cargo/struct.Workspace.html#method.partition_packages

Reply:
Yeah, I've seen this function & indend to use it, but as I've said this PR is in early stage. I've created it just to make sure other people know I'm working on it & wouldn't do duplicate work. Marked it draft & placed "work in progress" notice in the body. Anyway, thanks for your review comments.